plugins/jenkins-backend/src/plugin.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { loggerToWinstonLogger } from '@backstage/backend-common';
+import {
+  createBackendPlugin,
+  coreServices,
+} from '@backstage/backend-plugin-api';
+import { DefaultJenkinsInfoProvider } from './service/jenkinsInfoProvider';
+import { createRouter } from './service/router';
+import { CatalogClient } from '@backstage/catalog-client';
+
+/**
+ * Jenkins backend plugin
+ *
+ * @public
+ */
+
+export const jenkinsPlugin = createBackendPlugin({
+  pluginId: 'jenkins',
+  register(env) {
+    env.registerInit({
+      deps: {
+        logger: coreServices.logger,
+        permissions: coreServices.permissions,
+        httpRouter: coreServices.httpRouter,
+        config: coreServices.rootConfig,
+      },
+
+      async init({ logger, permissions, httpRouter, config }) {
+        const winstonLogger = loggerToWinstonLogger(logger);
+        const catalogBaseUrl = config.getString('catalog.baseUrl');
+        const discoveryApi = {
+          getBaseUrl: async () => catalogBaseUrl,
+        };
+        const catalogClient = new CatalogClient({ discoveryApi });
```

Comment:
I can follow the idea, but there is an easier & recommended way to get the Catalog here (e.g. https://github.com/backstage/backstage/blob/7f3adafa1495916fa2bad42f656c3d7a5ce019f2/plugins/lighthouse-backend/src/plugin.ts#L34-L52) 

There maybe instances, which don't use the Catalog Plugin. Therefor we want to make sure to mark it as a dependency.

Reply:
Thanks for the sample code, I have updated